### DISCUSSION TITLE
What colour patterns are supported for (label)color parameter?

### DISCUSSION THREAD

**Discussion by Andre601** [badges/shields#7128] [Q&A] [answered]
The site mentions that hex, rgb, rgba, hsl, hsla and CSS named colour values are supported. My question here now is: What are the different patterns?

I do know for a fact that hex can be `#rrggbb` or just `rrggbb` but what if I want to use rgba or hsla? How is the pattern there?

Additionally, would I like to know, if the badges support completely transparent colours to just display text within a document?

**Comment by chris48s** [comment, marked as answer]
https://github.com/badges/shields/tree/master/badge-maker#colors may be helpful

In general you should be able to use any valid CSS colour, so rgb, rgba, hsl, hsla etc should all work fine e.g:
https://img.shields.io/badge/foo-bar-hsl%28270%2C60%25%2C70%25%29
https://img.shields.io/badge/foo-bar-hsla%28240%2C%20100%25%2C%2050%25%2C%201%29
https://img.shields.io/badge/foo-bar-rgb%28255%2C%200%2C%20153%2C%201%29
https://img.shields.io/badge/foo-bar-rgba%2851%2C%20170%2C%2051%2C%20%201%29
although the URLs are not super readable.

Transparency should also work fine too. For example:
https://img.shields.io/badge/foo-bar-hsl(270%2C%2060%25%2C%2050%25%2C%2010%25)

**Reply by Andre601** [reply]
Ah, so rgb(a) and hsl(a) need to be in the formats `rgb(r,g,b)` / `rgba(r,g,b,a)` and `hsl(h,s,l)` / `hsla(h,s,l,a)` respectively.
This should be mentioned somewhere to be honest as I had no clue that RGB(A) had a more specific syntax than hex.

I assumed it was `https://img.shields.io/badge/foo-bar-r,g,b`
